frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/SelectBox.jsx

```diff
@@ -197,11 +197,11 @@ function DataSourceSelect({
     closePopup && !isMulti && closePopup();
   };
 
-  let optionsCount = options.length;
+  let optionsCount = options?.length;
 
-  options.forEach((item) => {
-    if (item.options && item.options.length > 0) {
-      optionsCount += item.options.length;
+  options?.forEach((item) => {
+    if (item?.options && item?.options?.length > 0) {
+      optionsCount += item.options?.length;
```

Comment:
Since validation is done on line 203,  optional chaining would not be needed here wdyt ?